hikari/impl/entity_factory.py

```diff
@@ -1709,6 +1711,7 @@ def deserialize_command(
         return commands.Command(
             app=self._app,
             id=snowflakes.Snowflake(payload["id"]),
+            type=commands.CommandType(payload.get("type", 0)),
```

Comment:
This is hardcoding the value, which we dont really want to do. Additionally, dont they always send it?

Reply:
Docs say it may not be sent. Although it's supposed to default to 1 and I somehow managed to put 0 here which ain't right.

Reply:
What do you mean by hardcoding in this context? The casting to an enum?

Reply:
You hardcode the integer, which may not be an issue now, but can be confusing for maintaining in the future. You can have a walrus operator to get whatever is in the payload and, if none, use the enum object as a default, instead of casting. This way we can find the uses later if we ever need it

Reply:
As in `type=payload.get("type", commands.CommandType.CHAT_INPUT)`?

Reply:
I meant more along the lines:

```py
type = CommandType(raw_type) if (raw_type := payload.get("type")) else CommandType.CHAT_INPUT
```

Or what you said, but then it would be:

```py
type = CommandType(payload.get("type", CommandType.CHAT_INPUT))
```


Not really sure what the best would be, maybe the second as you said.

Maybe as an alternative would be to ask Discord to always send it